lib/src/common/model/repos_statuses.dart

```diff
@@ -2,62 +2,83 @@ part of github.common;
-  String sha;
-  int totalCount;
-  List<RepositoryStatus> statuses;
-  Repository repository;
+  final String state;
+  final String sha;
+  final int totalCount;
+  final List<RepositoryStatus> statuses;
+  final Repository repository;
 
-  CombinedRepositoryStatus();
+  CombinedRepositoryStatus._({
+    @required this.state,
+    @required this.sha,
+    @required this.totalCount,
+    @required this.statuses,
+    @required this.repository,
+  });
 
   static CombinedRepositoryStatus fromJSON(Map<String, dynamic> input) {
     if (input == null) return null;
 
-    return CombinedRepositoryStatus()
-      ..state = input["state"]
-      ..sha = input["sha"]
-      ..totalCount = input["total_count"]
-      ..statuses = (input["statuses"] as List<Map<String, dynamic>>)
+    return CombinedRepositoryStatus._(
+      state: input["state"],
+      sha: input["sha"],
+      totalCount: input["total_count"],
+      statuses: (input["statuses"] as List<Map<String, dynamic>>)
           .map((it) => RepositoryStatus.fromJSON(it))
-          .toList()
-      ..repository =
-          Repository.fromJSON(input["repository"] as Map<String, dynamic>);
+          .toList(),
+      repository:
+          Repository.fromJSON(input["repository"] as Map<String, dynamic>),
+    );
   }
 }
 
 /// Model class for the status of a repository at a particular reference.
 class RepositoryStatus {
-  DateTime createdAt;
-  DateTime updatedAt;
-  String state;
-  String targetUrl;
-  String description;
-  String context;
+  final DateTime createdAt;
+  final DateTime updatedAt;
+  final String state;
+  final String targetUrl;
+  final String description;
+  final String context;
+
+  RepositoryStatus._({
+    @required this.createdAt,
+    @required this.updatedAt,
+    @required this.state,
+    @required this.targetUrl,
+    @required this.description,
+    @required this.context,
+  });
 
-  static RepositoryStatus fromJSON(Map<String, dynamic> input) {
+  factory RepositoryStatus.fromJSON(Map<String, dynamic> input) {
     if (input == null) return null;
 
-    return RepositoryStatus()
-      ..createdAt = parseDateTime(input['created_at'])
-      ..updatedAt = parseDateTime(input['updated_at'])
-      ..state = input['state']
-      ..targetUrl = input['target_url']
-      ..description = input['description']
-      ..context = input['context'];
+    return RepositoryStatus._(
+      createdAt: parseDateTime(input['created_at']),
+      updatedAt: parseDateTime(input['updated_at']),
+      state: input['state'],
+      targetUrl: input['target_url'],
+      description: input['description'],
+      context: input['context'],
+    );
   }
 }
 
 /// Model class for a new repository status to be created.
+// TODO: this class is unused
 class CreateStatus {
   final String state;
 
   @JsonKey(name: "target_url")
-  String targetUrl;
+  final String targetUrl;
 
-  String description;
-  String context;
+  final String description;
+  final String context;
```

Comment:
Classes that start with `Create` are intended to be instantiated, values set, and sent to github to create something. These fields then need to be non-final so they can be set to something 
![image](https://user-images.githubusercontent.com/314858/66706770-5d1f0480-ecf4-11e9-8f37-9b39d284850f.png)

Reply:
I checked all the other Create classes and they are all ok.

Reply:
The constructor is intended to set these fields. 
I can't find the line on your screenshot?

Reply:
What I did was pull down your branch and try to construct a CreateStatus object. The constructor only takes `state` so I couldn't  construct it with any values except null for `targetUrl`, `description`,  and `context`. The screenshot demonstrates that once it's constructed you can't update the values. I would suggest restoring these 3 variables to not be final. It's possible that there is code out there that looks like my screenshot and would no longer work.

Reply:
Done!